src/services/WolfDen.Application/Requests/Commands/Attendence/StatusUpdation/StatusUpdateCommandHandler.cs

```diff
@@ -0,0 +1,126 @@
+
+                if (holiday.Type == "Normal")
+                {
+                    statusId = 5;
+                }
+                else
+                {
+                    //LeaveRequest leave = await _context.LeaveRequest.Where(x => x.EmpId == request.EmployeeId && x.FromDate == request.Date && x.status == 2).FirstOrDefaultAsync(cancellationToken);
+                    //if (leave != null)
+                    //{
+                    //    if (leave.TypeName == "WFH")
+                    //    {
+                    //        statusId = 3;
+                    //    }
+                    //    else
+                    //    {
+                    statusId = 5;
+                    //    }
+                    //}
+                }
+            }
+            else
+            {
+                var insideHrs = await _context.DailyAttendence.Where(x => x.EmployeeId == request.EmployeeId && x.Date == request.Date).Select(x => x.InsideDuration).FirstOrDefaultAsync(cancellationToken);
+                if (insideHrs < 360)
+                {
+                    statusId = 2;
+                }
+                else
+                {
+                    statusId = 1;
+                }
+            }
+
+            if (statusId == 2)
+            {
+                List<string> cCManagerEmail = new List<string>();
+
+                string[] recieverEmail = new string[1];
+
+                var recEmail = await _context.Employees.Where(x => x.Id == request.EmployeeId)
+               .Select(x => x.Email).FirstOrDefaultAsync(cancellationToken);
+
+                var recName = await _context.Employees.Where(x => x.Id == request.EmployeeId)
+                    .Select(x => x.FirstName + " " + x.LastName).FirstOrDefaultAsync(cancellationToken);
+
+                if (recEmail != null)
+                {
+                    recieverEmail[0] = recEmail;
+                }
+
+                var managerId = await _context.Employees.Where(x => x.Id == request.EmployeeId)
+                    .Select(x => x.ManagerId).FirstOrDefaultAsync();
+
+                if (managerId != null)
+                {
+                    cCManagerEmail = await FindManager(managerId, cancellationToken);
+                }
+
+                string subject = "Attendence Low Warning";
+                string message = $"{recName}'s working hour is less than required";
+                SendEmailCommand.SendMail(senderEmail, senderName, recieverEmail, message, subject, cCManagerEmail.ToArray());
+
+            }
+
+            Status status = new Status(request.EmployeeId, request.Date, statusId);
+
+            await _context.AddAsync(status);
+            await _context.SaveChangesAsync(cancellationToken);
+            return status.Id;
+
+        }
+        private async Task<List<string>> FindManager(int? managerId, CancellationToken cancellationToken)

```

Comment:
how will managerId be null for this function?

Reply:
if (manager.ManagerId != null)
  {
      var managerEmails=await FindManager(manager.ManagerId, cancellationToken);
      cCManagerEmail.AddRange(managerEmails);
  }
  
  here it is recursive ,the base condition is met when the manager id is null,error comes when it is given as 
   FindManager(int managerId...)